pkg/operator/quorumguardcleanup/quorumguardcleanupcontroller.go

```diff
@@ -0,0 +1,141 @@
+package quorumguardcleanup
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/kubernetes"
+	appsv1listers "k8s.io/client-go/listers/apps/v1"
+	corev1listers "k8s.io/client-go/listers/core/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog/v2"
+
+	"github.com/openshift/cluster-etcd-operator/pkg/operator/operatorclient"
+)
+
+const (
+	oldEtcdGuardDeploymentName    = "etcd-quorum-guard"
+	newGuardLabelSelectorString   = "app=guard"
+	masterNodeLabelSelectorString = "node-role.kubernetes.io/master"
+)
+
+// QuorumGuardCleanupController watches the old EtcdQuorumGuard deployment and removes it
+// when the new staticpod controller guard pods have been rolled out
+type QuorumGuardCleanupController struct {
+	operatorClient   v1helpers.OperatorClient
+	kubeClient       kubernetes.Interface
+	podLister        corev1listers.PodLister
+	deploymentLister appsv1listers.DeploymentLister
+	masterNodeLister corev1listers.NodeLister
+}
+
+func NewQuorumGuardCleanupController(
+	operatorClient v1helpers.OperatorClient,
+	kubeClient kubernetes.Interface,
+	kubeInformers v1helpers.KubeInformersForNamespaces,
+	masterNodeInformer cache.SharedIndexInformer,
+	eventRecorder events.Recorder,
+) factory.Controller {
+	c := &QuorumGuardCleanupController{
+		operatorClient:   operatorClient,
+		kubeClient:       kubeClient,
+		podLister:        kubeInformers.InformersFor(operatorclient.TargetNamespace).Core().V1().Pods().Lister(),
+		deploymentLister: kubeInformers.InformersFor(operatorclient.TargetNamespace).Apps().V1().Deployments().Lister(),
+		masterNodeLister: corev1listers.NewNodeLister(masterNodeInformer.GetIndexer()),
+	}
+	return factory.New().ResyncEvery(5*time.Minute).WithInformers(
+		kubeInformers.InformersFor(operatorclient.TargetNamespace).Core().V1().Pods().Informer(),
+		kubeInformers.InformersFor(operatorclient.TargetNamespace).Apps().V1().Deployments().Informer(),
+	).WithSync(c.sync).ToController("QuorumGuardCleanupController", eventRecorder.WithComponentSuffix("quorum-guard-cleanup-controller"))
+}
+
+func (c *QuorumGuardCleanupController) sync(ctx context.Context, syncCtx factory.SyncContext) error {
```

Comment:
You need a check to be sure that the current level of this clusteroperator's version matches the desired level.  Otherwise when this operator starts during an upgrade, it will immediately remove quorum guard protection.

Reply:
> You need a check to be sure that the current level of this clusteroperator's version matches the desired level. Otherwise when this operator starts during an upgrade, it will immediately remove quorum guard protection.

oh, I see it down below by looking for the actual pods.  I think that's more complicated, but if you prefer it, ok.